tests/L0/run_transformer/test_layers.py

```diff
@@ -171,139 +270,274 @@ def test_affine_weight_init_row_parallel_gpu(self) -> None:
         self._affine_weight_init_test_impl(init_device="gpu", is_column_parallel=False)
 
     def test_row_parallel_linear(self) -> None:
-        for tensor_model_parallel_world_size in range(1, self.world_size + 1):
+        self._row_parallel_linear_test_impl(False, False, False)
+
+    def test_row_parallel_linear_gradient_accumulation_fusion(self) -> None:
+        self._row_parallel_linear_test_impl(True, False, False)
+
+    def test_row_parallel_linear_gradient_accumulation_fusion_in_fp16(self) -> None:
+        self._row_parallel_linear_test_impl(True, True, False)
+
+    @unittest.skipIf(torch.cuda.device_count() < 2, "Sequence Parallel requires >=2 GPUs")
+    def test_row_parallel_linear_sequence_parallel(self) -> None:
+        self._row_parallel_linear_test_impl(False, False, True)
+
+    # TODO(mkozuki): Merge this with `_column_parallel_linear_test_impl`
+    # Note that `input_is_parallel` is unique to `RowParallelLinear` which could make the merge complicated.
+    def _row_parallel_linear_test_impl(
+        self,
+        gradient_accumulation_fusion: bool,
+        accumulation_in_fp16: bool,
+        sequence_parallel_enabled: bool,
+    ) -> None:
+        tensor_shape = (
+            self.SEQUENCE_LENGTH,
+            self.BATCH_SIZE,
+            self.HIDDEN_SIZE,
+        )
+        for tensor_model_parallel_world_size in range(
+            1 + int(sequence_parallel_enabled), self.world_size + 1
+        ):
             if self.world_size % tensor_model_parallel_world_size:
                 continue
+            if sequence_parallel_enabled and tensor_model_parallel_world_size == 1:
+                continue
```

Comment:
`test_row_parallel_linear_sequence_parallel` has a decorator which obviates this.
```suggestion
```